### ISSUE TITLE
Default value not extracted when coming from a dict

### ISSUE THREAD

**Issue by Thomzoy** [mkdocstrings/griffe#59]
**Describe the bug**
When defining  default values for parameters of a function from a dictionary, *Griffe* doesn't extract this default value:

**To Reproduce**
Project structure:
```
test_griffe
│   __init__.py
│   my_mod.py  
```

In `my_mod.py`:
```python
DEFAULT_DICT = dict(
    key1="value1",
)

DEFAULT_VALUE = 3


def f(
    param1: str = DEFAULT_DICT["key1"],
    param2: int = DEFAULT_VALUE,
):
    """
    A griffe test function

    Parameters
    ----------
    param1 : str, optional
        _description_, by default DEFAULT_DICT["key1"]
    param2 : int, optional
        _description_, by default DEFAULT_VALUE
    """
    return
```

Truncated output of `griffe test_griffe`
```json
          {
            "kind": "function",
            "name": "f",
            "lineno": 9,
            "docstring": {
              "value": "[THE DOCSTRING]",
              "lineno": 25,
              "endlineno": null
            },
            "labels": [],
            "members": [],
            "decorators": [],
            "parameters": [
              {
                "name": "param1",
                "annotation": {
                  "source": "str",
                  "full": "str"
                },
                "kind": "positional or keyword",
                "default": null
              },
              {
                "name": "param2",
                "annotation": {
                  "source": "int",
                  "full": "int"
                },
                "kind": "positional or keyword",
                "default": "DEFAULT_VALUE"
              }
            ],
            "returns": null
          }
```
**Expected behavior**
- The `"default"` for param2 is correct (although a side question might be, could it display `3`, the actual value stored in `DEFAULT_VALUE` ?)
- The default for `param1` is `null`, while I was expecting one of the following:
   - "DEFAULT_DICT['key1']
   - "value1"

Maybe I'm just not fiddling with the options enough ?

Thnaks for your help !

**System (please complete the following information):**
- `griffe` version: [0.14.1]
- Python version: [3.7]
- OS: [Linux]

**Comment by pawamoy** [comment]
Hey again @Thomzoy, thanks for the report!

> 
    The "default" for param2 is correct (although a side question might be, could it display 3, the actual value stored in DEFAULT_VALUE ?)

It cannot, and I'm really inclined to say that it must not. Most of the time you want to see the variable name, not the value, just like for enums. You can add a docstring to `DEFAULT_VALUE` and point to it in the function docstring with a cross-ref for example. Maybe names in defaults will automatically point to the related objects in the future.

>    The default for param1 is null, while I was expecting one of the following:
        "DEFAULT_DICT['key1']
        "value1"

Indeed, it's surely the same bug as in https://github.com/mkdocstrings/python/issues/8, as you mentioned in your PR.

**Comment by pawamoy** [comment]
Fixed in Griffe 0.14.1
